File: archive/cdk-ts/Pr1058/lib/networking-stack.ts
import * as cdk from 'aws-cdk-lib';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';

interface NetworkingStackProps extends cdk.StackProps {
  environmentSuffix: string;
}

export class NetworkingStack extends cdk.Stack {
  public readonly vpc: ec2.Vpc;
  public readonly privateSubnets: ec2.ISubnet[];
  public readonly publicSubnets: ec2.ISubnet[];

  constructor(scope: Construct, id: string, props: NetworkingStackProps) {
    super(scope, id, props);

    // Create VPC with public and private subnets across 2 AZs
    this.vpc = new ec2.Vpc(this, `secure-${props.environmentSuffix}-vpc`, {
      vpcName: `secure-${props.environmentSuffix}-vpc`,
      maxAzs: 2,
      natGateways: 2,
      subnetConfiguration: [
        {
          name: `secure-${props.environmentSuffix}-public`,
          subnetType: ec2.SubnetType.PUBLIC,
          cidrMask: 24,
        },
        {
          name: `secure-${props.environmentSuffix}-private`,
          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
          cidrMask: 24,
        },
        {
          name: `secure-${props.environmentSuffix}-isolated`,
          subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
          cidrMask: 24,
        },
      ],
      enableDnsHostnames: true,
      enableDnsSupport: true,
    });

    this.privateSubnets = this.vpc.privateSubnets;
    this.publicSubnets = this.vpc.publicSubnets;

    // Create strict Network ACLs for additional security
    const privateNacl = new ec2.NetworkAcl(
      this,
      `secure-${props.environmentSuffix}-private-nacl`,
      {
        vpc: this.vpc,
        networkAclName: `secure-${props.environmentSuffix}-private-nacl`,
      }
    );

    // Allow HTTPS outbound from private subnets
    privateNacl.addEntry(`secure-${props.environmentSuffix}-https-out`, {
      ruleNumber: 100,
      traffic: ec2.AclTraffic.tcpPort(443),
      direction: ec2.TrafficDirection.EGRESS,
      cidr: ec2.AclCidr.anyIpv4(),
    });

    // Allow HTTP outbound for package updates
    privateNacl.addEntry(`secure-${props.environmentSuffix}-http-out`, {
      ruleNumber: 110,
      traffic: ec2.AclTraffic.tcpPort(80),
      direction: ec2.TrafficDirection.EGRESS,
      cidr: ec2.AclCidr.anyIpv4(),
    });

    // Apply tags
    cdk.Tags.of(this).add('Environment', props.environmentSuffix);
    cdk.Tags.of(this).add('ProjectName', 'secure-infrastructure');
    cdk.Tags.of(this).add('CostCenter', 'security-team');
  }
}

File: archive/cdk-go/Pr2360/lib/networking_stack.go
package lib

import (
	"github.com/aws/aws-cdk-go/awscdk/v2"
	"github.com/aws/aws-cdk-go/awscdk/v2/awsec2"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type NetworkingStackProps struct {
	*awscdk.NestedStackProps
	EnvironmentSuffix *string
}

type NetworkingStack struct {
	awscdk.NestedStack
	Vpc awsec2.IVpc
}

func NewNetworkingStack(scope constructs.Construct, id *string, props *NetworkingStackProps) *NetworkingStack {
	nestedStack := awscdk.NewNestedStack(scope, id, props.NestedStackProps)

	// Get environment suffix with fallback
	envSuffix := "dev"
	if props != nil && props.EnvironmentSuffix != nil && *props.EnvironmentSuffix != "" {
		envSuffix = *props.EnvironmentSuffix
	}

	// Create VPC with public and private subnets
	vpc := awsec2.NewVpc(nestedStack, jsii.String("VPC"), &awsec2.VpcProps{
		MaxAzs:      jsii.Number(3),
		IpAddresses: awsec2.IpAddresses_Cidr(jsii.String("10.0.0.0/16")),
		SubnetConfiguration: &[]*awsec2.SubnetConfiguration{
			{
				CidrMask:   jsii.Number(24),
				Name:       jsii.String("Public"),
				SubnetType: awsec2.SubnetType_PUBLIC,
			},
			{
				CidrMask:   jsii.Number(24),
				Name:       jsii.String("Private"),
				SubnetType: awsec2.SubnetType_PRIVATE_WITH_EGRESS,
			},
			{
				CidrMask:   jsii.Number(28),
				Name:       jsii.String("Database"),
				SubnetType: awsec2.SubnetType_PRIVATE_ISOLATED,
			},
		},
		NatGateways:        jsii.Number(2), // For high availability
		EnableDnsHostnames: jsii.Bool(true),
		EnableDnsSupport:   jsii.Bool(true),
	})

	// Add tags to VPC
	awscdk.Tags_Of(vpc).Add(jsii.String("Name"), jsii.String("tap-vpc-"+envSuffix), nil)

	return &NetworkingStack{
		NestedStack: nestedStack,
		Vpc:         vpc,
	}
}
